plugins/repository-hdfs/src/test/java/org/elasticsearch/repositories/hdfs/HdfsBlobStoreContainerTests.java

```diff
@@ -18,16 +18,78 @@
  */
 package org.elasticsearch.repositories.hdfs;
 
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.AbstractFileSystem;
 import org.apache.hadoop.fs.FileContext;
+import org.apache.hadoop.fs.UnsupportedFileSystemException;
 import org.elasticsearch.common.blobstore.BlobStore;
 import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
 
+import javax.security.auth.Subject;
 import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.Principal;
+import java.security.PrivilegedAction;
+import java.util.Collections;
 
 public class HdfsBlobStoreContainerTests extends ESBlobStoreContainerTestCase {
 
     @Override
     protected BlobStore newBlobStore() throws IOException {
-        return new HdfsBlobStore(FileContext.getFileContext(), "", 100);
+        Path tempDir = createTempDir(null);
+        FileContext fileContext = AccessController.doPrivileged(
+                new PrivilegedAction<FileContext>() {
+                    @Override
+                    public FileContext run() {
+                        return createContext(tempDir.toUri());
+            }
+        });
+
+        return new HdfsBlobStore(fileContext, "temp", 100);
+    }
+
+    public FileContext createContext(URI uri) {
+        Configuration cfg = new Configuration(true);
+        cfg.setClassLoader(HdfsRepository.class.getClassLoader());
+        cfg.reloadConfiguration();
```

Comment:
@costin @jbaiera Do you have any insights as to what should go in the configuration properties for this test to allow it to create an HDFS FileContext for a temporary directory location (created via `createTempDir()`) for this unit test?  This `createContext` method is missing these lines that occur in `HdfsRepository`, but I'm not sure what values should go into them:
```
Map<String, String> map = repositorySettings.settings().getByPrefix("conf.").getAsMap();
        for (Entry<String, String> entry : map.entrySet()) {
            cfg.set(entry.getKey(), entry.getValue());
        }```